java-api-core/src/com/kinvey/java/store/requests/data/save/SaveRequest.kt

```diff
@@ -71,12 +74,12 @@ class SaveRequest<T : GenericJson>(private val cache: ICache<T>?, private val ne
                         // to remove the entity from the local cache with the temporary ID.
                         cache?.delete(id)
                     }
+                    ret?.let { cache?.save(it) }
                 } catch (e: IOException) {
                     syncManager?.enqueueRequest(networkManager?.collectionName,
                     networkManager, if (bRealmGeneratedId == true) SyncRequest.HttpVerb.POST else SyncRequest.HttpVerb.PUT, id)
-                    throw e
+                    if (e is HttpResponseException && e.statusCode == 401) throw e
```

Comment:
let's wrap the original exception inside another exception object so we don't lose the original stack details.
something like `throw new IOException(e)`, I think that's the Java way of doing this since Java does not have a better way to rethrow an exception like C# does